src/response_sender.cc

```diff
@@ -147,22 +147,38 @@ ResponseSender::Send(
   }
 
   if (has_gpu_output) {
+    ScopedDefer _([send_message_payload] {
+      bi::scoped_lock<bi::interprocess_mutex> guard{send_message_payload->mu};
+      send_message_payload->is_stub_turn = false;
+      send_message_payload->cv.notify_one();
+      while (!send_message_payload->is_stub_turn) {
+        // Wait for the stub process to send the response and populate error
+        // message if any.
+        send_message_payload->cv.wait(guard);
+      }
+    });
+
     AllocatedSharedMemory<GPUBuffersShm> gpu_buffers_handle =
         shm_pool_->Load<GPUBuffersShm>(
             send_message_payload->gpu_buffers_handle);
+    if (!gpu_buffers_handle.data_->success) {
```

Comment:
Great enhancement for adding error handling in GPU buffer loading. Do we also have any test coverage for this part?

Reply:
Yes, the existing [test_dlpack_tensor_error()](https://github.com/triton-inference-server/server/blob/main/qa/L0_backend_python/python_test.py#L206) will test this part. The test currently test the non-decoupled data pipeline, once this is merged, it will test the decoupled data pipeline which requires this change to pass.